src/main/java/seedu/address/model/Statistics.java

```diff
@@ -23,33 +28,120 @@ public Statistics(ReadOnlyStatistics toBeCopied) {
         resetData(toBeCopied);
     }
 
+    /** Initialises empty DayData for past MAX_SIZE days */
+    public void init() {
+        LocalDate currDate = LocalDate.now();
+        for (int i = 0; i < MAX_SIZE; i++) {
+            LocalDate tempLocalDate = currDate.minusDays(i);
+            String tempLocalDateStr = tempLocalDate.toString();
+            Date tempDate = new Date(tempLocalDateStr);
+            DayData dayData = new DayData(tempDate);
+            dayDataList.add(dayData);
+        }
+    }
+
     //// list overwrite operations
-    /**
-     * Replaces the contents of the person list with {@code persons}. {@code persons} must not
-     * contain duplicate persons.
-     */
+
+    /** Clears list */
+    public void clearList() {
+        this.dayDataList.clear();
+    }
+
+    /** Replaces the contents of the list with {@code dayDataList}. */
     public void setDayDatas(List<DayData> dayDataList) {
         this.dayDataList.clear();
         this.dayDataList.addAll(dayDataList);
     }
 
-    /** Resets the existing data of this {@code TaskList} with {@code newData}. */
+    /** Resets the existing data of this {@code Statistics} with {@code newData}. */
     public void resetData(ReadOnlyStatistics newData) {
         requireNonNull(newData);
         setDayDatas(newData.getDayDataList());
     }
 
-    //// person-level operations
+    //// dayData-level operations
+
+    // FUNCTIONS FOR
+    // HARDOHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH SIR
+
+    /** reinitialises dayDataList to current day while retaining stored data. */
+    public void fetch() {
+        HashMap<Date, PomDurationData> pomTempStorage = new HashMap<>();
```

Comment:
Actually not too sure why this function is needed, I was thinking to manage the max 7 rolling dates in statistics, it would be a queue? then we just keep editing the tail of the queue for new stats and poll and enqueue when the tail of the queue is not today's date

Reply:
Yup this makes sense I wasn't thinking straight, thanks for the suggestion!